### PR TITLE
Google analytics v4: Oauth2.0 config

### LINKED ISSUE #6173: Source Google Analytics: Support webflow Oauth
## Tell us about the problem you're trying to solve
We currently only support json_key authentication to Google Analytics. However, with the release of Airbyte Cloud, we need to start supporting Oauth, the more common way of authenticating to GA. 

Instead of the current `credentials_json` field the GA config should accept one of: 
1. `credentials_json`
2. `oauth` <-- this one should accept client_id, client_secret, and refresh token

This should be a oneof nested inside a top-level field called "auth_mechanism" . See the connector spec reference in the docs for reference on how this can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying the json key in the config's top-level should not be impacted by this change.

### PR DESCRIPTION
**Pull request by avida**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
Resolves #6173

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 1]
```diff
@@ -4,14 +4,40 @@
...
     "properties": {
-      "credentials_json": {
-        "type": "string",
-        "title": "Credentials JSON",
-        "description": "The contents of the JSON service account key. Check out the <a href=\"https://docs.airbyte.io/integrations/sources/googleanalytics\">docs</a> if you need help generating this key.",
-        "airbyte_secret": true
+      "auth_mechanism": {
+        "title": "Authentication mechanism",
+        "type": "object",
+        "description": "Chose either OAuth2.0 flow or provide your own JWT credentials for service account",
```
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:13]:
```suggestion
        "description": "Choose either OAuth2.0 flow or provide your own JWT credentials for service account",
```

Comment (avida) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:13]: Fixed

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 2]
```diff
@@ -4,14 +4,48 @@
...
-    "required": ["credentials_json", "view_id", "start_date"],
+    "required": ["view_id", "start_date"],
     "additionalProperties": false,
     "properties": {
-      "credentials_json": {
-        "type": "string",
-        "title": "Credentials JSON",
-        "description": "The contents of the JSON service account key. Check out the <a href=\"https://docs.airbyte.io/integrations/sources/googleanalytics\">docs</a> if you need help generating this key.",
-        "airbyte_secret": true
+      "auth_mechanism": {
```
[the remark this answers is not in the document]
Comment (avida) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:10]: Done

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 3]
```diff
@@ -4,14 +4,48 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Google Analytics V4 Spec",
     "type": "object",
-    "required": ["credentials_json", "view_id", "start_date"],
+    "required": ["view_id", "start_date"],
```
[the remark this answers is not in the document]
Comment (avida) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:7]: fixed

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 4]
```diff
@@ -4,14 +4,48 @@
...
-      "credentials_json": {
-        "type": "string",
-        "title": "Credentials JSON",
-        "description": "The contents of the JSON service account key. Check out the <a href=\"https://docs.airbyte.io/integrations/sources/googleanalytics\">docs</a> if you need help generating this key.",
-        "airbyte_secret": true
+      "auth_mechanism": {
+        "title": "Authentication mechanism",
+        "type": "object",
+        "description": "Choose either OAuth2.0 flow or provide your own JWT credentials for service account",
+        "oneOf": [
```
[the remark this answers is not in the document]
Comment (avida) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:14]: Done

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 5]
```diff
@@ -4,14 +4,48 @@
...
+              "client_id": { "type": "string" },
+              "client_secret": { "type": "string", "airbyte_secret": true },
+              "refresh_token": { "type": "string", "airbyte_secret": true }
+            },
+            "required": ["client_id", "client_secret", "refresh_token"],
+            "additionalProperties": false
+          },
+          {
+            "type": "object",
+            "title": "JWT authorization",
```
[the remark this answers is not in the document]
Comment (avida) [reply airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:32]: Done

**Review by Zirochkaa — APPROVED** [review]

**Pull request merged**